### PR TITLE
allow for default authenticated user

### PR DESCRIPTION
**Pull request by aaronmallen**
### Added
- the ability to define a default authenticated user.

### CODE REVIEW

**Comment by aaronmallen** [comment]
tested and working :D

**Comment by aaronmallen** [comment]
Derps fixed (I think) and again tested and again working :D

File: programs/github [hunk 1]
```diff
@@ -176,12 +184,12 @@ if action == subcommands.auth then
 	local user, token = args[1], args[2]
 	if not user then return printError("No user specified.") end
 
-	if flags.delete then
+	if flags.remove then
 		github.Auth.delete(user)
 		print(('Deleted github token for user %s'):format(user))
 	else
 		if not token then return printError("No token specified.") end
```
Comment (eric-wieser) [inline programs/github]: Should be possible to run something like `github auth eric --set-default` without a token

File: programs/github [hunk 2]
```diff
@@ -16,10 +16,12 @@ local subcommands = {
 		}
 	},
 	auth = {
-		usage = "github auth <user> [<api token> | -d ]",
+		usage = "github auth <user> [<api token> | -r | -d]",
```
Comment (eric-wieser) [inline programs/github]: `github auth <user> (<api token> [-r] | -r | -d)` seems more accurate

File: programs/github (hunk 1 above)
Comment (aaronmallen) [inline programs/github]: Oh I like that!

File: programs/github [hunk 3]
```diff
@@ -16,10 +16,12 @@ local subcommands = {
...
 		flags = {
-			['-d'] = 'delete',
-			['--delete'] = 'delete'
+			['-r'] = 'remove',
+			['--remove'] = 'remove',
+			['-d'] = 'default',
```
Comment (aaronmallen) [inline programs/github]: Fair, we can def use a more verbose flag for setting a default

**Comment by aaronmallen** [comment]
So my only (loose) argument for not AUTOMATICALLY specifying a default is every request where am Auth flag isn't specified is going to be made with that user's token, that to me should be more of an opt-in so the user knows EXACTLY what's going on.  If I'm on a sever and sharing a computer with Steve, I may not want Steve cloning repos as me. (sorry for any grammar/spelling I'm mobile)

File: programs/github (hunk 2 above)
Comment (aaronmallen) [inline programs/github]: With the recommended changes wouldn't this be `github Auth <user> (<api token> [-d | - - set-default]) `

**Comment by aaronmallen** [comment]
@eric-wieser bump

**Comment by eric-wieser** [comment]
Am I right in thinking these are the full set of commands this adds? Not sure if this supports the third one, but I think it maybe should?

 * `github auth <user> <token>`
 * `github auth <user> --set-default`
 * `github auth <user> <token> --set-default`
 * `github auth <user> -d`

**Pull request closed without merging**